src/shared/clients/devenvClient.ts

```diff
@@ -103,6 +108,67 @@ export class DevEnvClient implements vscode.Disposable {
         // `Authorization` _should_ have two parameters (RFC 7235), MDE should probably fix that
         headers: { Authorization: this.authToken },
     })
+
+    #lastKnownActivityTimestamp: number | undefined
+
+    async updateUserActivity(timestamp: number = Date.now()): Promise<number> {
+        await this.got.put('activity', { json: { timestamp: timestamp.toString() } })
+        this.#lastKnownActivityTimestamp = timestamp
+        this.#updatedActivityEventEmitter.fire(timestamp)
+        return timestamp
+    }
+
+    async getUserActivity(): Promise<number | undefined> {
+        const response = await this.got<GetActivityResponse>('activity')
+        const latestTimestamp = response.body.timestamp ? parseInt(response.body.timestamp) : undefined
+
+        // A single Dev Env can have multiple clients connected to it.
+        // So if one client updates the timestamp, it will be different from what the
+        // other clients assumed the last activity was.
+        if (latestTimestamp && latestTimestamp !== this.#lastKnownActivityTimestamp) {
```

Comment:
```suggestion
        if (latestTimestamp && latestTimestamp > this.#lastKnownActivityTimestamp) {
```

Reply:
It is possible an older timestamp is set, so the `latestTimestamp` could be older than the one we previously knew about.